File: src/main/java/Deleter.java
import javax.swing.JFrame;
import javax.swing.JList;
import javax.swing.JScrollPane;
import javax.swing.UIManager;
import javax.swing.UnsupportedLookAndFeelException;
import javax.swing.WindowConstants;
import java.io.File;
import java.io.IOException;
import java.util.ArrayList;

public class Deleter {
    private int tries = 0;
    private final JList<String> text = new JList<>(new String[]{"Deleter starting..."});

    private Deleter() {}

    public static void main(String[] args) throws UnsupportedLookAndFeelException, ClassNotFoundException, InterruptedException, InstantiationException, IllegalAccessException, IOException {
        Deleter deleter = new Deleter();
        deleter.run(args);
    }

    private void run(String[] files) throws UnsupportedLookAndFeelException, ClassNotFoundException, InstantiationException, IllegalAccessException, InterruptedException, IOException {
        append("--- Files to delete :");
        for (String arg : files) {
            append("    " + arg);
        }
        append("---------------------");

        JFrame frame = new JFrame("Deleter");
        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
        JScrollPane scroll = new JScrollPane(text);
        scroll.setBounds(0, 0, 500, 500);
        frame.add(scroll);
        frame.setSize(500, 500);
        frame.setVisible(true);
        frame.setFocusable(true);
        frame.setAlwaysOnTop(true);
        frame.setLocationRelativeTo(null);
        frame.setResizable(false);
        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        for (String file : files) {
            tries = 0;
            delete(new File(file));
        }
        append("Done! You can now close this window.");
        Thread.sleep(50000);
        System.exit(0);
    }

    private void delete(File file) throws InterruptedException {
        append("Checking if \"" + file.getAbsolutePath() + "\" exists");
        if (!file.exists()) {
            append("\"" + file.getAbsolutePath() + "\" does not exist!");
            return;
        }

        append("Deleting file \"" + file.getAbsolutePath() + "\"");

        if (!file.delete()) {
            tries += 1;
            if (tries >= 4) {
                append("\nToo many tries for \"" + file.getAbsolutePath() + "\", cancelling.");
                return;
            }
            append("\n\"" + file.getAbsolutePath() + "\" failed to delete, trying again in " + tries + " seconds...");
            Thread.sleep(1000L * tries);
            delete(file);
        } else {
            append("Successfully deleted \"" + file.getAbsolutePath() + "\"!");
        }
    }

    private void append(String string) {
        System.out.println(string);
        ArrayList<String> list = getList();
        list.add(string);
        text.setModel(new JListList(list));
    }

    private ArrayList<String> getList() {
        ArrayList<String> list = new ArrayList<>();
        int i = 0;
        while (i < text.getModel().getSize()) {
            i++;
            list.add(text.getModel().getElementAt(i - 1));
        }
        return list;
    }
}

File: src/main/kotlin/Deleter.kt
import java.io.File
import java.io.PrintStream
import javax.swing.*
import javax.swing.event.EventListenerList
import javax.swing.event.ListDataListener
import kotlin.system.exitProcess


var tries = 0
val text = JList(arrayOf("Deleter starting..."))

val logfile = File("skyclientupdater/files/deleterlog.txt")

fun main(files : Array<String>) {
    append("+++ ARGS")
    append(files.joinToString())
    append("--- ARGS")

    val frame = JFrame("Deleter 1.7")
    UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName())
    val scroll = JScrollPane(text)
    scroll.setBounds(0, 0, 500, 500)
    frame.add(scroll)
    frame.setSize(500, 500)
    frame.isVisible = true
    frame.isFocusable = true
    frame.isAlwaysOnTop = true
    frame.setLocationRelativeTo(null)
    frame.isResizable = false
    frame.defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
    for (file in files) {
        tries = 0
        delete(File(file))
    }
    Thread.sleep(1000)
    exitProcess(0)
}

private fun delete(file: File) {
    append("checking if \"${file.absolutePath}\" exists")
    if (!file.exists()) {
        append("\"${file.absolutePath}\" does not exist!")
        return
    }

    append("deleting file \"${file.absolutePath}\"")

    if (!file.delete()) {
        tries += 1
        if (tries >= 4) {
            append("\nToo many tries for \"${file.absolutePath}\", cancelling.")
            return
        }
        append("\n\"${file.absolutePath}\" failed to delete, trying again in $tries seconds...")
        Thread.sleep((1000 * tries).toLong())
        delete(file)
    } else {
        append("Successfully deleted \"${file.absolutePath}\"!")
    }
}

private fun append(string: String) {
    logfile.appendText(string + "\n")
    println(string)
    text.model = JListList(getList().also { it.add(string) })
}

private fun getList(): ArrayList<String> {
    val list = arrayListOf<String>()
    var i = 0
    while (i < text.model.size) {
        i++
        list.add(text.model.getElementAt(i - 1))
    }
    return list
}

class JListList(private val list: List<String>): ListModel<String> {
    private var listeners = EventListenerList()
    override fun getSize(): Int = list.size

    override fun getElementAt(index: Int): String = list[index]

    override fun addListDataListener(l: ListDataListener) {
        listeners.add(ListDataListener::class.java, l)
    }

    override fun removeListDataListener(l: ListDataListener) {
        listeners.remove(ListDataListener::class.java, l)
    }
}
